api/layout/paths.go

```diff
@@ -29,20 +29,32 @@ const (
 // EntriesPathForLogIndex builds the local path at which the leaf with the given index lives in.
 // Note that this will be an entry bundle containing up to 256 entries and thus multiple
 // indices can map to the same output path.
-//
-// TODO(mhutchinson): revisit to consider how partial tile suffixes should be added.
-func EntriesPathForLogIndex(seq uint64) string {
-	return EntriesPath(seq / 256)
+// The logSize is required so that a partial qualifier can be appended to tiles that
+// would contain fewer than 256 entries.
+func EntriesPathForLogIndex(seq, logSize uint64) string {
+	p := partialTileSize(0, seq, logSize)
+	return EntriesPath(seq/256, p)
 }
 
-// EntriesPath returns the local path for the Nth entry bundle.
-func EntriesPath(N uint64) string {
-	return fmt.Sprintf("tile/entries%s", fmtN(N))
+// EntriesPath returns the local path for the nth entry bundle. p denotes the partial
+// tile size, or 0 if the tile is complete.
+func EntriesPath(n, p uint64) string {
```

Comment:
Should we make this take `logSize` too for consistency?

Reply:
I had some reason for doing it this way, but I think that reason is bogus. I'll take a look at this.

Reply:
Done.